### PR TITLE
[Fixes #40] Suppress caching for ErrorHandler

### LINKED ISSUE #40: Suppress caching for ErrorHandler
When using the ErrorHandler to re-execute a failed request and generate a page, that error page should not be cached. However, components like the static file middleware cache by default via e-tags.

Consider modifying the samples to manually add no-cache headers on the app-error path, or modify the middleware itself to set these no-cache headers. before re-executing the request.

### PR DESCRIPTION
**Pull request by kichalla**
@Tratcher

### CODE REVIEW

File: src/Microsoft.AspNet.Diagnostics/ErrorHandlerMiddleware.cs [hunk 1]
```diff
@@ -70,9 +74,17 @@ public ErrorHandlerMiddleware(RequestDelegate next, ILoggerFactory loggerFactory
...
                 throw; // Re-throw the original if we couldn't handle it
             }
         }
+
+        private Task ClearCacheHeaders(object state)
+        {
+            var response = (HttpResponse)state;
+            response.Headers.Set("Cache-Control", "no-cache");
+            response.Headers.Set("Pragma", "no-cache");
+            response.Headers.Set("Expires", "-1");
```
Comment (kichalla) [inline src/Microsoft.AspNet.Diagnostics/ErrorHandlerMiddleware.cs]: I suppose I should not be doing anything regarding ETag headers?

Comment (Tratcher) [inline src/Microsoft.AspNet.Diagnostics/ErrorHandlerMiddleware.cs]: I'd recommend removing the e-tag header as well.

Comment (kichalla) [inline src/Microsoft.AspNet.Diagnostics/ErrorHandlerMiddleware.cs]: Updated. cc: @Tratcher

**Comment by Tratcher** [comment]
Add a few tests and :shipit:

**Comment by kichalla** [comment]
@Tratcher added tests for this. can you take a look? (especially about the new project.json references)

File: test/Microsoft.AspNet.Diagnostics.Tests/project.json [hunk 2]
```diff
@@ -4,6 +4,9 @@
     },
     "dependencies": {
         "Microsoft.AspNet.Diagnostics.Elm": "1.0.0-*",
+        "Microsoft.AspNet.FileProviders.Embedded": "1.0.0-*",
+        "Microsoft.AspNet.StaticFiles": "1.0.0-*",
```
Comment (Tratcher) [inline test/Microsoft.AspNet.Diagnostics.Tests/project.json]: Middleware repos should avoid taking dependencies on other middleware repos. It complicates the CI dependency chain. You should be able to mimic the behavior of StaticFiles accurately enough for testing.

Comment (kichalla) [inline test/Microsoft.AspNet.Diagnostics.Tests/project.json]: Sure, I have removed the test specifically targeting StaticFiles as this scenario is already being covered in the other test that I added. Also added new tests. @Tratcher

File: test/Microsoft.AspNet.Diagnostics.Tests/ErrorHandlerTest.cs [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+                app.Run(async (httpContext) =>
+                {
+                    await httpContext.Response.WriteAsync("Hello");
+                    throw new InvalidOperationException("Something bad happened");
+                });
+            }))
+            {
+                var client = server.CreateClient();
+                var response = await client.GetAsync(string.Empty);
```
Comment (Tratcher) [inline test/Microsoft.AspNet.Diagnostics.Tests/ErrorHandlerTest.cs]: Doesn't the TestServer throw the InvalidOperationException back to you here?

**Comment by Tratcher** [comment]
squash and :shipit:

**Pull request merged**